omlmd/helpers.py

```diff
@@ -95,14 +92,20 @@ def push(
         ]
         try:
             # print(target, files, model_metadata.to_annotations_dict())
-            result = self._registry.push(
+            result = self.registry.push(
                 target=target,
                 files=files,
                 manifest_annotations=model_metadata.to_annotations_dict(),
                 manifest_config=manifest_cfg,
                 do_chunked=True,
             )
-            self.notify_listeners(PushEvent(target, model_metadata))
+            self.notify_listeners(
+                PushEvent(
+                    result.headers["Docker-Content-Digest"],
```

Comment:
do you think we should safeguard against errors here? IMO this is a panic

Reply:
mmm, fair point.

I would say, let's use `result` as the first element of the PushEvent tuple. This way, user can decide how they want to handle if server result is an 4xx or 5xx

On the PushEvent dataclass, then add another method ~`digest()` which will check the result is a 2xx and extract this `Docker-Content-Digest` header, throwing an error if this header is missing in case of 2xx

wdyt ?